source/base/MessageHandler.h

```diff
@@ -20,9 +20,11 @@ class MessageHandler : public QObject
 
     void setupWithClient(Swift::Client* client);
 
-    void sendMessage(QString const &toJid, QString const &message, QString const &type, bool isGroup);
+    void sendMessage(QString const &toJid, QString const &message, QString const &type, bool isGroup, QString const &MsgId="");
     void sendDisplayedForJid(const QString &jid);
 
+    void saveMessageToSendLater(QString const &toJid, QString const &message, QString const &type, QString const &msgId);
```

Comment:
Not found in cpp. You add the logic to inject the msg in the db direct in Shmoose.cpp. I would prefer it to be inside MessageHandler. But I will add an extra discussion on that later.

Reply:
Sorry. This must be removed. I tested several solutions before the final implementation.